app/models/info_request/batch_pagination.rb

```diff
@@ -0,0 +1,22 @@
+# Methods to handle pagination within an associated InfoRequestBatch
+module InfoRequest::BatchPagination
+  def next_in_batch
+    return nil unless info_request_batch
+    info_request_batch.info_requests[index_in_batch + 1] || first_in_batch
+  end
+
+  def prev_in_batch
+    return nil unless info_request_batch
+    info_request_batch.info_requests[index_in_batch - 1]
+  end
+
+  private
+
+  def first_in_batch
+    info_request_batch.info_requests.first
+  end
+
+  def index_in_batch
+    info_request_batch.info_requests.index(self)
```

Comment:
Potentially loading 500 `InfoRequest` objects, pluck the IDs and find the index of `self.id`?